### DISCUSSION TITLE
Neuralforecasting pytorch dependency issues

### DISCUSSION THREAD

**Discussion by egolub22** [Nixtla/neuralforecast#1044] [Q&A]
I am attempting to utilize the neuralforecasting package as part of my ml solution but I am getting a series of pytorch related dependency errors that are preventing me from using it. The first of these errors appears when I try to initially import the package. 

`
Import statements:

    from neuralforecast import NeuralForecast
    from neuralforecast.models import LSTM, NHITS, RNN
    from neuralforecast.auto import AutoNHITS
    from neuralforecast.losses.pytorch import MAE
    from neuralforecast.losses.pytorch import MAPE
    
    import plotly
    from plotly.subplots import make_subplots
    import plotly.graph_objs as go
    from utilsforecast.losses import *
    
    print(version('plotly'))
    print(torch.__version__)
`

Error:
<img width="678" alt="Import_error_2" src="https://github.com/Nixtla/mlforecast/assets/43010144/a8ee9385-0fb0-4595-b0ff-ec2be42d7f1e">

I fixed this using the terminal comand: pip install neuralforecast "torch<2.0.0", which was suggested in another online forum. I am not sure if this is the source of the following issues. The next issue I encounter occures when I try to instantiate an mlforecast object.

`
Code:

    horizon = 31
    models = [LSTM(h=horizon,                    # Forecast horizon
                   max_steps=500,                # Number of steps to train
                   scaler_type='standard',       # Type of scaler to normalize data
                   encoder_hidden_size=64,       # Defines the size of the hidden state of the LSTM
                   decoder_hidden_size=64,),     # Defines the number of hidden units of each layer of the MLP decoder
              NHITS(h=horizon,                   # Forecast horizon
                    input_size=2 * horizon,      # Length of input sequence
                    max_steps=100,               # Number of steps to train
                    n_freq_downsample=[2, 1, 1]) # Downsampling factors for each stack output
              ]
    nf = NeuralForecast(models=models, freq='D')
    nf.fit(df=df)
`

Error:
<img width="1590" alt="Instantiation_error" src="https://github.com/Nixtla/mlforecast/assets/43010144/a398e5de-b919-45a1-bb60-53b492f9e575">

I'm not sure if there is a certain version of torch/pytorch that is recommended for the neuralforecast package but I have tried upgrading those packages to various version as well as the neuralforcasting package itself to eliminate the error to no avail.

**Comment by jmoralez** [comment]
You need pytorch_lightning>=2.1.0

**Reply by egolub22** [reply]
upgrading pytorch_lightning to 2.1.0 still gives me the first error about disable_possible_user_warnings

**Reply by jmoralez** [reply]
It's there https://github.com/Lightning-AI/pytorch-lightning/blob/2.1.0/src/lightning/fabric/utilities/warnings.py#L26
